river/tree/mondrian/mondrian_tree_nodes.py

```diff
@@ -366,3 +366,150 @@ class MondrianBranchClassifier(MondrianNodeClassifier, MondrianBranch):
 
     def __init__(self, parent, time, depth, feature, threshold, *children):
         super().__init__(parent, time, depth, feature, threshold, *children)
+
+
+class MondrianNodeRegressor(MondrianNode):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.n_samples = 0
+        self.mean = 0.0
+
+    def replant(self, leaf: MondrianNodeRegressor, copy_all: bool = False):
+        """Transfer information from a leaf to a new branch."""
+        self.weight = leaf.weight  # type: ignore
+        self.log_weight_tree = leaf.log_weight_tree  # type: ignore
+        self.mean = leaf.mean
+
+        if copy_all:
+            self.memory_range_min = leaf.memory_range_min
+            self.memory_range_max = leaf.memory_range_max
+            self.n_samples = leaf.n_samples
+
+    def predict(self) -> base.typing.RegTarget:
+        """Return the prediction of the node."""
+        return self.mean
+
+    def loss(self, sample_value: base.typing.RegTarget) -> float:
+        """Compute the loss of the node.
+
+        Parameters
+        ----------
+        sample_value
+            A given value.
+        """
+
+        r = self.predict() - sample_value
```

Comment:
Is this the correct order in the regression case?

Reply:
I think so ! It is squared anyways so I don't know if it changes much. 

(https://github.com/onelearn/onelearn/blob/master/onelearn/node_methods.py)